lib/scholar/covariance/ledoit_wolf.ex

```diff
@@ -182,9 +149,9 @@ defmodule Scholar.Covariance.LedoitWolf do
 
   defnp ledoit_wolf_shrinkage_complex(x) do
     {num_samples, num_features} = Nx.shape(x)
-    emp_cov = empirical_covariance(x)
+    emp_cov = Nx.covariance(x)
```

Comment:
Is `x` centered here? If yes, maybe reverting it to `empirical_covariance` might be better. 😅

Reply:
Hmm, I don't think `empirical_covariance/1` was centering `x,` if that's what you're asking about. However, I'm still a bit confused as to why, when I used `empirical_covariance/1` instead of `Nx.covariance/2,` the `Covariance.*.fit/2` functions produced the same output as the scikit-learn versions when I passed a non-centered `x` and set `assume_centered?` to `true.` While I feel this case is invalid and can't think of any practical use case for passing such arguments, it still feels odd that the output in this edge case differs when using `Nx.covariance/2` compared to the scikit-learn version. It seems `empirical_covariance` might not operate exactly the same way as `Nx.covariance/2` , so I also like the idea of reverting it to `empirical_covariance`.

Reply:
Lemme have a look.

Reply:
I think you need 
```suggestion
    emp_cov = Nx.covariance(x, Nx.reshape(0, {1}))
```

for the code to be totally equivalent.

empirical_covariance was blindly assuming the mean is 0

Reply:
> empirical_covariance was blindly assuming the mean is 0

Yes, because the data should be centered before. That is why using `empirical_covariance` might be better idea. Like this the data won't be centered twice.

Reply:
So you want me to change it back to `empirical_covariance` version?

Reply:
@norm4nn let's revert to `empirical_covariance` for now so we can merge this and get unstuck. :)

Reply:
Sure, done